chromium_src/chrome/browser/accessibility_ui.cc

```diff
@@ -0,0 +1,442 @@
+  data.SetString(kHTML, is_web_enabled ? (html ? kOn : kOff) : kDisabled);
+
+  // TODO(codebytere): enable use of this flag.
+  //
+  // The "labelImages" flag works only if "web" is enabled, the current profile
+  // has the kAccessibilityImageLabelsEnabled preference set and the appropriate
+  // command line switch has been used. Since this is so closely tied into user
+  // prefs and causes bugs, we're disabling it for now.
+  bool are_accessibility_image_labels_enabled = is_web_enabled;
+  data.SetString(kLabelImages, kDisabled);
+
+  // The "pdf" flag is independent of the others.
+  data.SetString(kPDF, pdf ? kOn : kOff);
+
+  // Always dump the Accessibility tree.
+  data.SetString(kInternal, kOn);
+
+  std::unique_ptr<base::ListValue> rvh_list(new base::ListValue());
+  std::unique_ptr<content::RenderWidgetHostIterator> widgets(
+      content::RenderWidgetHost::GetRenderWidgetHosts());
+
+  while (content::RenderWidgetHost* widget = widgets->GetNextHost()) {
+    // Ignore processes that don't have a connection, such as crashed tabs.
+    if (!widget->GetProcess()->IsInitializedAndNotDead())
+      continue;
+    content::RenderViewHost* rvh = content::RenderViewHost::From(widget);
+    if (!rvh)
+      continue;
+    content::WebContents* web_contents =
+        content::WebContents::FromRenderViewHost(rvh);
+    content::WebContentsDelegate* delegate = web_contents->GetDelegate();
+    if (!delegate)
+      continue;
+    // Ignore views that are never user-visible, like background pages.
+    if (delegate->IsNeverComposited(web_contents))
+      continue;
+    content::BrowserContext* context = rvh->GetProcess()->GetBrowserContext();
+    if (context != current_context)
+      continue;
+
+    std::unique_ptr<base::DictionaryValue> descriptor =
+        BuildTargetDescriptor(rvh);
+    descriptor->SetBoolean(kNative, is_native_enabled);
+    descriptor->SetBoolean(kWeb, is_web_enabled);
+    descriptor->SetBoolean(kLabelImages,
+                           are_accessibility_image_labels_enabled);
+    rvh_list->Append(std::move(descriptor));
+  }
+
+  data.Set(kPagesField, std::move(rvh_list));
+
+  v8::Isolate* isolate = electron::JavascriptEnvironment::GetIsolate();
+  v8::HandleScope handle_scope(isolate);
+
+  std::unique_ptr<base::ListValue> window_list(new base::ListValue());
+  for (auto val : electron::api::BaseWindow::GetAll(isolate)) {
+    electron::api::BaseWindow* base_window;
+    if (!gin::Converter<electron::api::BaseWindow*>::FromV8(isolate, val,
+                                                            &base_window))
+      return;
+    window_list->Append(BuildTargetDescriptor(base_window));
+  }
+
+  data.Set(kBrowsersField, std::move(window_list));
+
+  std::string json_string;
+  base::JSONWriter::Write(data, &json_string);
+
+  std::move(callback).Run(base::RefCountedString::TakeString(&json_string));
+}
+
+}  // namespace
+
+ElectronAccessibilityUI::ElectronAccessibilityUI(content::WebUI* web_ui)
```

Comment:
Can you make `shouldHandleAccessibilityRequestCallback` and `HandleAccessibilityRequestCallback` members of `AccessibilityUI`, so that we just call them and edit the results, also eliminates some more utility functions in this file.

```
void HandleAccessibilityRequestCallback(
    content::BrowserContext* current_context,
    const std::string& path,
    content::WebUIDataSource::GotDataCallback callback) {
  auto json_data = AccessibilityUI::HandleAccessibilityRequestCallback(current_context, path);
  // Update the kBrowsersField in json_data

  std::string json_string;
  base::JSONWriter::Write(json_data, &json_string);

  std::move(callback).Run(base::RefCountedString::TakeString(&json_string));
}
```

In AccessibilityUI make 

```
void HandleAccessibilityRequestCallback(
    content::BrowserContext* current_context,
    const std::string& path,
    content::WebUIDataSource::GotDataCallback callback) {}
```

to 

```
base::DictionaryValue HandleAccessibilityRequestCallback(
    content::BrowserContext* current_context,
    const std::string& path) {  }
```